### PR TITLE
Various small bugfixes

### PR DESCRIPTION
**Pull request by FFroehlich**
1) fixes an issue with startpoint resampling
2) removes redundant computation of lower order derivatives for AmiciObjective
3) adds some debug logging to amici objective

### CODE REVIEW

**Comment by FFroehlich** [comment]
@yannikschaelte could you please have a look at this?

File: pypesto/objective/amici_objective.py [hunk 1]
```diff
@@ -84,17 +79,31 @@ def sres(x):
         # extract parameter names from model
         self.x_names = list(self.amici_model.getParameterNames())
 
+    def fun(self, x):
+        return self.call_amici(
+            x,
+            Objective.MODE_FUN
+        )
+
+    def res(self, x):
```
[the remark this answers is not in the document]
Comment (yannikschaelte) [reply pypesto/objective/amici_objective.py]: is this overwriting of fun, res necessary? codacy complains and it might be error-prone and is also not so nice to have ``overwritefun`` in the constructor. i think it is valid to save the sensi_orders temporarily, as you did in the Objective. But maybe one can just have in ``__init__``
```
def fun(x):
    return self._call_amici(x, Objective.MODE_FUN)
...
super().__init__(...fun=fun,...)
```
and similarly for res?

File: pypesto/objective/amici_objective.py [hunk 2]
```diff
@@ -162,20 +179,18 @@ def call_amici(
...
-        return Objective.map_to_output(
-            sensi_orders,
-            mode,
-            fval=nllh, grad=snllh, hess=ssnllh,
-            res=res, sres=sres
-        )
+        if mode == Objective.MODE_FUN:
+            return nllh, snllh, ssnllh
+        elif mode == Objective.MODE_RES:
+            return res, sres
```
Comment (yannikschaelte) [inline pypesto/objective/amici_objective.py]: why is this here? i think fun, i.e. here _call_amici, should return just the values as a tuple that were asked for in tuple form, so e.g. fval if sensi_orders is (0,), or (fval, grad) if (0,1). now always 3 resp. 2 values are returned, looks like this migt lead to errors when using e.g. derivative-free optimizers. does this really work? haven't checked, can do that when i have time.

Comment (FFroehlich) [reply pypesto/objective/amici_objective.py]: `_call_amici` implements `fun` and `res` which are supposed to return the outputs exactly in that order. The subindexing according to the specified `sensi_orders` is applied in `_call_mode_fun` and `_call_mode_res` and `map_to_output`.

File: pypesto/objective/objective.py [hunk 3]
```diff
@@ -244,21 +264,34 @@ def _update_trace(self, x, mode, result):
...
+                else res_result
+            sres = None if not self.options.trace_record_res \
+                else sres_result
+            chi2 = None if not self.options.trace_record_chi2 \
+                           or res_result is None \
+                else np.inner(res_result,res_result)/2
+            schi2 = None if not self.options.trace_record_chi2 \
+                            or sres_result is None \
+                            or res_result is None \
+                else res_result * res_result
```
Comment (yannikschaelte) [inline pypesto/objective/objective.py]: for chi2, can't we use the res_to_fval function? then the normalization is always the same. is the formula for schi2 correct? res_result * res_sresult maybe?

File: pypesto/objective/objective.py [hunk 4]
```diff
@@ -646,8 +734,10 @@ def map_to_output(sensi_orders, mode, **kwargs):
                 output += (kwargs[Objective.RES],)
             if 1 in sensi_orders:
                 output += (kwargs[Objective.SRES],)
-        if len(output) == 1:
-            # return a single value not as tuple
+        if len(output) == 1 and (
```
Comment (yannikschaelte) [inline pypesto/objective/objective.py]: are these 2 additional checks really needed?

Comment (FFroehlich) [reply pypesto/objective/objective.py]: yep :(
